File: Python/MultiThreaded/crdFiles.py
import random
import time
import os
import threading

def generateRandList(size):
  L = []
  for i in range(0, size):
    L.append(random.randint(0, 32767))
  return L

def writeFile(filename, contents):
  with open(filename, 'w') as file:
    file.write(contents)

def readFile(filename):
  with open(filename, 'r') as file:
    return file.read()
  
def createReadDelete(filename, contents):
  writeFile(filename, contents)
  readFile(filename)
  os.remove(filename)

if __name__ == "__main__":
  # Setup and Generate Random number lists
  RandList = generateRandList(16)
  threads = []

  # Perform selection sorts and measure time
  start = time.perf_counter()
  for i in range(0, 16):
    name = f"randomFile{i}.txt"
    th = threading.Thread(target=createReadDelete, args=(name, str(RandList[i])))
    th.start()
    threads.append(th)
  for th in threads:
    th.join()
  stop = time.perf_counter()
  duration = stop - start

  # Print Results
  print(f'{duration:5f}')

File: C#/MultiThreaded/crdFiles.cs
using System;
using System.Diagnostics;
using System.Collections.Generic;
using System.IO;
using System.Threading;

class Program
{
  static Random random = new Random();

  static List<int> generateRandList(int size) {
    List<int> L = new List<int>(size);
    for (int i = 0; i < size; i++) {
      L.Add(random.Next(0, 32767));
    }
  return L;
  }

  static void writeFile(string filename, string contents) {
        using (StreamWriter writer = new StreamWriter(filename))
        {
            writer.WriteLine(contents); // Writes the string to the file
        }
  }
  
  static string readFile(string filename) {
        using (StreamReader reader = new StreamReader(filename))
        {
            string line;
            line = reader.ReadLine();
            return line;
        } 
  }

  static void createReadDelete(string filename, string contents) {
    writeFile(filename, contents);
    readFile(filename);
    File.Delete(filename) ;
  }

  static void Main()
  {
    // Setup and Generate Random number list
    List<int> RandList = generateRandList(16);
    Stopwatch stopwatch = new Stopwatch();
    List<Thread> threads = new List<Thread>();

    // Perform file operations and measure time
    stopwatch.Start();
    for (int i=0; i < RandList.Count; i++) {
      string name = $"randomFile{i}.txt";
      string content = RandList[i].ToString();
      Thread th = new Thread(() => createReadDelete(name, content));
      th.Start();
      threads.Add(th);
    }
    for (int i=0; i<16; i++){
        threads[i].Join();
    }
    stopwatch.Stop();
    TimeSpan duration = stopwatch.Elapsed;
    double durationSecs = duration.TotalMilliseconds/1000;
    string durationStr = durationSecs.ToString("0.000000");

    // Print Results
    Console.WriteLine(durationStr);
  }
}